lib/solver.ml

```diff
@@ -73,9 +73,9 @@ let solve_sub expr sol ty1 ty2 =
   let ty1' = lookup_type sol ty1 in
   let ty2' = lookup_type sol ty2 in
   let inf = inf ty1' ty2' in
-  if is_subtype ty1' ty2' then
+  if is_subtype ty1' ty2' && ty1' <> ty2' then
```

Comment:
It is my understanding that this change aim to add new type variables (i.e., type variables which are not appeared in `sol` yet) into `sol`.
Is it correct?

If so, for example, assume `ty1` has type `number()` in `sol` and `ty2` is not included in sol, `ty2` will not be added to `sol`. Is it OK?

And, I would appreciate that you add some comments about the purpose because I am worrying that we may forget meaning of `&& ty1' <> ty2'` in the future.

Reply:
> It is my understanding that this change aim to add new type variables (i.e., type variables which are not appeared in `sol` yet) into `sol`.
> Is it correct?

Yes it is.

> If so, for example, assume `ty1` has type `number()` in `sol` and `ty2` is not included in sol, `ty2` will not be added to `sol`. Is it OK?

It is no problem that variables in right hand side are not restricted in the solving algorithms.

Reply:
> And, I would appreciate that you add some comments about the purpose because I am worrying that we may forget meaning of `&& ty1' <> ty2'` in the future.

Sure it is, I will fix it.

Reply:
> It is no problem that variables in right hand side are not restricted in the solving algorithms.

I understand. Thanks!